dashboard/app/models/user.rb

```diff
@@ -836,7 +836,7 @@ def User.track_proficiency(user_id, script_id, level_id)
 
   # returns whether a new level has been completed and asynchronously enqueues an operation
   # to update the level progress.
-  def track_level_progress_async(script_level, new_result, submitted)
+  def track_level_progress_async(script_level, new_result, submitted, level_source_id)
```

Comment:
I'm a little torn on this.  One one hand, default `nil` reduces the diff.  On the other hand I want to make sure callers know about this param and pass it in when it's available.  Let me think about this a little more.